src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java

```diff
@@ -101,4 +106,22 @@
         .discoveryApiName(discoveryApiName)
         .hasMultipleServices(false);
   }
+
+  private List<PackageDependencyView> getDependencies(
+      Map<String, VersionBound> dependencies, Set<String> whitelistedDependencies) {
+    List<PackageDependencyView> protoPackageDependencies = new ArrayList<>();
+    if (dependencies != null) {
+      for (Map.Entry<String, VersionBound> entry : dependencies.entrySet()) {
+        if (entry.getValue() != null
+            && (whitelistedDependencies == null
+                || whitelistedDependencies.contains(entry.getKey()))) {
```

Comment:
Apart from null checks, I think this can be simpler written (partly-pseudocode)
```java
Map dep = new Map(dependencies); // make copy
dep.keySet().retainAll(whitelistedDependencies);
for (entry : deps.entrySet()) {
  list.add(PackageDependencyView.create(...);
}
```